### ISSUE TITLE
sparkConf metrics definitions fail due to aggressive wildcard verification

### ISSUE THREAD

**Issue by lexpierce** [intel-hadoop/HiBench#352]
```
❯ bin/workloads/micro/sort/prepare/prepare.sh
patching args=
Parsing conf: /home/alex/Source/HiBench/conf/hadoop.conf
Parsing conf: /home/alex/Source/HiBench/conf/hibench.conf
Parsing conf: /home/alex/Source/HiBench/conf/spark.conf
Parsing conf: /home/alex/Source/HiBench/conf/workloads/micro/sort.conf
Traceback (most recent call last):
  File "/home/alex/Source/HiBench/bin/functions/load_config.py", line 723, in <module>
    load_config(conf_root, workload_configFile, workload_folder, patching_config)
  File "/home/alex/Source/HiBench/bin/functions/load_config.py", line 206, in load_config
    waterfall_config()
  File "/home/alex/Source/HiBench/bin/functions/load_config.py", line 322, in waterfall_config
    key, value):     # not wildcard rules? re-add
  File "/home/alex/Source/HiBench/bin/functions/load_config.py", line 263, in wildcard_replacement
    ), "ERROR! The number of wildcards defined in key and value must be identical: %s -> %s" % (key, value)
AssertionError: ERROR! The number of wildcards defined in key and value must be identical: spark.metrics.conf.*.source.jvm.class -> org.apache.spark.metrics.source.JvmSource
/home/alex/Source/HiBench/bin/functions/workload-functions.sh: line 38: .: filename argument required
```

With spark-1.5 and newer, this is a supported way to define spark metrics sinks... see [SPARK-7169](https://issues.apache.org/jira/browse/SPARK-7169)

**Comment by carsonwang** [comment]
@lexpierce , thanks for trying the 6.0 version and report the issue. We'll investigate and fix it.

**Comment by gczsjdy** [comment]
@lexpierce The exception is due to wrongly treat every line of config file with "*" with wildcard. I have created a [PR-358](https://github.com/intel-hadoop/HiBench/pull/358) to solve the problem.

**Comment by carsonwang** [comment]
Resolved by #358
